### ISSUE TITLE
Hearts of Iron III (25890)

### ISSUE THREAD

**Issue by Dazarlin** [ValveSoftware/Proton#889]
The game is read as "Running" for a second or two before closing with nothing actually opening.

Here's the log:
[steam-25890.log](https://github.com/ValveSoftware/Proton/files/2330871/steam-25890.log)

Here's my system info: https://gist.github.com/Dazarlin/1fe5b3d929f71f1f3f598c5b99d0c262

**Comment by thunder1410** [comment]
With the latest beta of proton I got into the launcher which had the web preview missing (internet explorer related?). When I pressed launch I got the loading screen for less then a second and then it crashed.
With lutris and winetrick I got it stuck at the same point, but after some more trying I ran coudln't get in the launcher anymore.
system information: https://gist.github.com/thunder1410/76897c9bd211c74b30968baa4037ce96

**Comment by ldenti** [comment]
I followed this guide on the [paradox forum](https://forum.paradoxplaza.com/forum/index.php?threads/howto-hoi3-linux-with-wine.425470/) (all the credit goes to that user) and then I was able to run the game without any problem (I went through the tutorial and I started a new game). These are the steps I followed:
1. Install the game (I have the base game plus all DLCs)
2. Run the game and let the First-Time Setup install the prerequisites. The game will crash
3. Run
```
WINEPREFIX=${path_to_steamlibrary}/steamapps/compatdata/25890/pfx/ winetricks directx9 dotnet20
```

4. Check the Window Version and set it to `Windows XP` with 
```
WINEPREFIX=${path_to_steamlibrary}/steamapps/compatdata/25890/pfx/ winecfg
```

5. Create the key `HKEY_CURRENT_USER/Software/Wine/Direct3D` and add the following string values: `OffscreenRenderingMode` with value *fbo* and `VideoMemorySize` with your GPU's memory in `regedit`:
```
WINEPREFIX=${path_to_steamlibrary}/steamapps/compatdata/25890/pfx/ wine regedit
```

6. Always in regedit, create the key `HKEY_CURRENT_USER/Software/Wine/DirectSound` and add the string value `MaxShadowSize` and set it to *0*

**Comment by lenzls** [comment]
@ldenti 
How did you manage to install `dotnet20`? 
It seems that the steam wineprefix is 64bit and the package doesn't support that.

```
[0] $> WINEPREFIX=~/.steam/steam/steamapps/compatdata/25890/pfx/ winetricks dotnet20
------------------------------------------------------
You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Using winetricks 20180217 - sha256sum: 1b156c616174f41df79d72a90c52eb2b110c63a6e8ecb865d303a8f9f0908924 with wine-3.0-rc6 and WINEARCH=win64
Executing w_do_call dotnet20
Executing load_dotnet20 
------------------------------------------------------
This package (dotnet20) does not work on a 64-bit installation. You must use a prefix made with WINEARCH=win32.

```


When I'm running the game from steam it always tries to install the net framework 2 with some wizard but fails immediately after clicking next with "This product is already installed as part of the operating system".
And for some reason the application does not continue launching but stops.

**Comment by tobiasheineken** [comment]
@lenzls I managed to run hoi3 on my Xubuntu 18.04, mostly following these instructions.

A few noteworthy points:

1. The forum suggests downloading `winetricks` directly instead of using `apt install`.
 This was important for me as `winetricks` in apt is too old for 64-bit dotnet20.
 I used https://wiki.ubuntuusers.de/winetricks/ to download a modern `winetricks` instead.

2. When calling `winetricks dotnet20`, the DotNet Installer asked me to Repair or uninstall dotnet. I chose repair and it worked.

3. After Step 6, I had to call 
```
WINEPREFIX=${path_to_steamlibrary}/steamapps/compatdata/25890/pfx/ winecfg
```
again and go to the `libraries` tab and remove `devenum.dll` and `quartz.dll`

After this, the game runs fine. I played a few hours of Soviet Union and it seems to work.

**Comment by lenzls** [comment]
@tobiasheineken 
I'm now a step further, since installing dotnet20 now worked with the newer winetricks as you said.

But mine still crashes immediately after startup. 

<details><summary>Error message looks sadly quite uninformative:</summary><p>

```
Unhandled exception: page fault on read access to 0x00000000 in 32-bit code (0x00abba63).
Register dump:
 CS:0023 SS:002b DS:002b ES:002b FS:0063 GS:006b
 EIP:00abba63 ESP:0032f4d4 EBP:0032f5c8 EFLAGS:00210287(  R- --  I S - -P-C)
 EAX:00000000 EBX:00000000 ECX:02732090 EDX:00000001
 ESI:0032f4e0 EDI:00000010
Stack dump:
0x0032f4d4:  027215a8 0032f748 00000000 015fce98
0x0032f4e4:  015a7be4 02731e08 00000000 00000000
0x0032f4f4:  0032f4ec 0032f4f0 00000000 00000000
0x0032f504:  0032f4fc 0032f500 00000000 00000000
0x0032f514:  0032f50c 0032f510 02732900 00000000
0x0032f524:  00000004 0032f58c 02232390 022347c8
Backtrace:
=>0 0x00abba63 in hoi3_tfh (+0x6bba63) (0x0032f5c8)
  1 0x00630b79 in hoi3_tfh (+0x230b78) (0x0032f930)
  2 0x00a5a02f in hoi3_tfh (+0x65a02e) (0x0032fe30)
  3 0x00b98834 in hoi3_tfh (+0x798833) (0x0032fec0)
  4 0x7b469cc2 call_process_entry+0x11() in kernel32 (0x0032fed8)
  5 0x7b46b51e in kernel32 (+0x4b51d) (0x0032ffd8)
  6 0x7b469cce call_process_entry+0x1d() in kernel32 (0x0032ffec)
0x00abba63: movl	0x0(%eax),%ecx
Modules:
Module	Address			Debug info	Name (112 modules)
PE	  330000-  336000	Deferred        lua51
PE	  340000-  369000	Deferred        lua5.1
PE	  370000-  396000	Deferred        tbb
PE	  3e0000-  3f2000	Deferred        tbbmalloc
PE	  400000- 1cae000	Export          hoi3_tfh
PE	10000000-101e5000	Deferred        d3dx9_42
PE	3b400000-3b41f000	Deferred        steam_api
ELF	7a800000-7a939000	Deferred        opengl32<elf>
  \-PE	7a820000-7a939000	\               opengl32
ELF	7b400000-7b7f4000	Dwarf           kernel32<elf>
  \-PE	7b420000-7b7f4000	\               kernel32
ELF	7bc00000-7bcfe000	Deferred        ntdll<elf>
  \-PE	7bc10000-7bcfe000	\               ntdll
ELF	7c000000-7c003000	Deferred        <wine-loader>
ELF	7c4d2000-7c4de000	Deferred        libpciaccess.so.0
ELF	7c4de000-7c4fc000	Deferred        libgcc_s.so.1
ELF	7c4fc000-7c506000	Deferred        libdrm_nouveau.so.2
ELF	7c506000-7c52d000	Deferred        libdrm_intel.so.1
ELF	7c52d000-7d285000	Deferred        i965_dri.so
ELF	7d285000-7d2a2000	Deferred        libxcb-glx.so.0
ELF	7d2a2000-7d2c2000	Deferred        libglapi.so.0
ELF	7d2c2000-7d33a000	Deferred        libglx_mesa.so.0
ELF	7d375000-7d3d4000	Deferred        libgldispatch.so.0
ELF	7d3d4000-7d435000	Deferred        libgl.so.1
ELF	7d536000-7d544000	Deferred        libdrm_radeon.so.1
ELF	7d564000-7d587000	Deferred        libglx.so.0
ELF	7d58b000-7d58e000	Deferred        libxshmfence.so.1
ELF	7d58e000-7d5a2000	Deferred        libdrm.so.2
ELF	7d664000-7d66c000	Deferred        libxcb-sync.so.1
ELF	7d66c000-7d670000	Deferred        libxcb-present.so.0
ELF	7d670000-7d675000	Deferred        libxcb-dri3.so.0
ELF	7d6bb000-7d6d3000	Deferred        userenv<elf>
  \-PE	7d6c0000-7d6d3000	\               userenv
ELF	7d6d3000-7d6da000	Deferred        libxfixes.so.3
ELF	7d6da000-7d6e6000	Deferred        libxcursor.so.1
ELF	7d6e6000-7d6f9000	Deferred        libxi.so.6
ELF	7d6f9000-7d6fd000	Deferred        libxcomposite.so.1
ELF	7d6fd000-7d70a000	Deferred        libxrandr.so.2
ELF	7d70a000-7d716000	Deferred        libxrender.so.1
ELF	7d716000-7d71d000	Deferred        libxxf86vm.so.1
ELF	7d71d000-7d721000	Deferred        libxinerama.so.1
ELF	7d721000-7d73c000	Deferred        libbsd.so.0
ELF	7d73c000-7d743000	Deferred        libxdmcp.so.6
ELF	7d743000-7d747000	Deferred        libxau.so.6
ELF	7d747000-7d773000	Deferred        libxcb.so.1
ELF	7d773000-7d8bd000	Deferred        libx11.so.6
ELF	7d8bd000-7d8d2000	Deferred        libxext.so.6
ELF	7d8d4000-7d8d8000	Deferred        libxdamage.so.1
ELF	7d8d8000-7d8de000	Deferred        libxcb-dri2.so.0
ELF	7d8de000-7d8e1000	Deferred        libx11-xcb.so.1
ELF	7d8e1000-7d90b000	Deferred        libpng12.so.0
ELF	7d90d000-7d9a1000	Deferred        winex11<elf>
  \-PE	7d920000-7d9a1000	\               winex11
ELF	7d9a1000-7d9c6000	Deferred        imm32<elf>
  \-PE	7d9b0000-7d9c6000	\               imm32
ELF	7db42000-7db74000	Deferred        libexpat.so.1
ELF	7db74000-7dbbf000	Deferred        libfontconfig.so.1
ELF	7dbbf000-7dbde000	Deferred        libz.so.1
ELF	7dc19000-7dcb5000	Deferred        libfreetype.so.6
ELF	7dcb5000-7dce1000	Deferred        msacm32<elf>
  \-PE	7dcc0000-7dce1000	\               msacm32
ELF	7dce1000-7dd9c000	Deferred        winmm<elf>
  \-PE	7dcf0000-7dd9c000	\               winmm
ELF	7dd9c000-7ddb7000	Deferred        dinput8<elf>
  \-PE	7dda0000-7ddb7000	\               dinput8
ELF	7ddb7000-7ddf5000	Deferred        ws2_32<elf>
  \-PE	7ddc0000-7ddf5000	\               ws2_32
ELF	7ddf5000-7de7d000	Deferred        rpcrt4<elf>
  \-PE	7de00000-7de7d000	\               rpcrt4
ELF	7de7d000-7dfe7000	Deferred        ole32<elf>
  \-PE	7dea0000-7dfe7000	\               ole32
ELF	7dfe7000-7e033000	Deferred        dsound<elf>
  \-PE	7dff0000-7e033000	\               dsound
ELF	7e033000-7e18b000	Deferred        wined3d<elf>
  \-PE	7e040000-7e18b000	\               wined3d
ELF	7e18b000-7e1cb000	Deferred        d3d9<elf>
  \-PE	7e190000-7e1cb000	\               d3d9
ELF	7e1cb000-7e2a6000	Deferred        msvcr100<elf>
  \-PE	7e1f0000-7e2a6000	\               msvcr100
ELF	7e2a6000-7e3d2000	Deferred        msvcp100<elf>
  \-PE	7e2f0000-7e3d2000	\               msvcp100
ELF	7e3d2000-7e44c000	Deferred        shlwapi<elf>
  \-PE	7e3e0000-7e44c000	\               shlwapi
ELF	7e44c000-7e715000	Deferred        shell32<elf>
  \-PE	7e460000-7e715000	\               shell32
ELF	7e715000-7e72f000	Deferred        version<elf>
  \-PE	7e720000-7e72f000	\               version
ELF	7e72f000-7e943000	Deferred        user32<elf>
  \-PE	7e740000-7e943000	\               user32
ELF	7e943000-7ea05000	Deferred        msvcr80<elf>
  \-PE	7e960000-7ea05000	\               msvcr80
ELF	7ea05000-7ea85000	Deferred        advapi32<elf>
  \-PE	7ea10000-7ea85000	\               advapi32
ELF	7ea85000-7ebba000	Deferred        gdi32<elf>
  \-PE	7ea90000-7ebba000	\               gdi32
ELF	7ebba000-7ec72000	Deferred        msvcrt<elf>
  \-PE	7ebd0000-7ec72000	\               msvcrt
ELF	7ee72000-7ee86000	Deferred        libnss_files.so.2
ELF	7ee86000-7eea1000	Deferred        libnsl.so.1
ELF	7eea1000-7eeaf000	Deferred        libnss_nis.so.2
ELF	7eeaf000-7eeb9000	Deferred        libnss_compat.so.2
ELF	7eeb9000-7efbb000	Deferred        libm.so.6
ELF	7efbb000-7efc5000	Deferred        librt.so.1
ELF	7efc7000-7efda000	Deferred        psapi<elf>
  \-PE	7efd0000-7efda000	\               psapi
ELF	f7bc7000-f7bcc000	Deferred        libdl.so.2
ELF	f7bcc000-f7da8000	Deferred        libc.so.6
ELF	f7da8000-f7dc7000	Deferred        libpthread.so.0
ELF	f7dc7000-f7f7e000	Dwarf           libwine.so.1
ELF	f7f7e000-f7f87000	Deferred        libgtk3-nocsd.so.0
ELF	f7fc4000-f7fec000	Deferred        ld-linux.so.2
ELF	f7fef000-f7ff1000	Deferred        [vdso].so
Threads:
process  tid      prio (all id:s are in hex)
00000008 (D) C:\Program Files\Hearts of Iron 3\hoi3_tfh.exe
	00000009    0 <==
0000000e services.exe
	00000029    0
	00000024    0
	0000001e    0
	0000001b    0
	00000013    0
	00000010    0
	0000000f    0
00000011 mscorsvw.exe
	00000018    0
	00000017    0
	00000016    0
	00000012    0
00000019 winedevice.exe
	00000021    0
	0000001d    0
	0000001c    0
	0000001a    0
00000022 plugplay.exe
	00000026    0
	00000025    0
	00000023    0
00000027 winedevice.exe
	0000002b    0
	0000002a    0
	00000028    0
0000002e explorer.exe
	00000032    0
	00000031    0
	00000030    0
	0000002f    0
System information:
    Wine build: wine-3.13
    Platform: i386
    Version: Windows XP
    Host system: Linux
    Host version: 4.15.0-88-generic

```

</p></details>

**Comment by tobiasheineken** [comment]
@lenzls What precisely do you mean by "after startup"?
I'm able to get to the main menu now (game itself, not just launcher), and I'm able to start the soviet campaign when Barbarossa starts.

I'm not able to start Japan or Germany. Which is - weird. But I'm working on it :)

**Comment by mineturtle36** [comment]
Applied all of this (and more!) from whatever I could find on the web. Same end result - crash after a few seconds of intro music and the first image shown (basically same as what lenzis and others here have been reporting). Tried both latest and the previous version of Proton.

Running latest Manjaro.

**Comment by yanivkfir** [comment]
I am having the same issue as mineturtle36 and lezis.  I remember the game did work on wine about two years ago (when I used Mint, and since it ran on wine it should probably run on proton) but I tried it against now with Manjaro and it doesn't work, neither on Wine or Proton. I suspect that it has something to do with the OS difference, probably some library version or something like that.

**Comment by thiras** [comment]
I can confirm the problem. I've tried all the guides I've found on the internet even ones almost 10 years old.

<details>
<summary>Here is my system info;</summary>

```

Computer Information:
    Manufacturer:  Unknown
    Model:  Unknown
    Form Factor: Laptop
    No Touch Input Detected

Processor Information:
    CPU Vendor:  GenuineIntel
    CPU Brand:  Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    CPU Family:  0x6
    CPU Model:  0x9e
    CPU Stepping:  0x9
    CPU Type:  0x0
    Speed:  3800 Mhz
    8 logical processors
    4 physical processors
    HyperThreading:  Supported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported
    AES:  Supported
    AVX:  Supported
    AVX2:  Unsupported
    AVX512F:  Unsupported
    AVX512PF:  Unsupported
    AVX512ER:  Unsupported
    AVX512CD:  Unsupported
    AVX512VNNI:  Unsupported
    SHA:  Unsupported
    CMPXCHG16B:  Supported
    LAHF/SAHF:  Supported
    PrefetchW:  Unsupported

Operating System Version:
    Ubuntu 20.04.1 LTS (64 bit)
    Kernel Name:  Linux
    Kernel Version:  5.4.0-42-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  12008000
    X Window Manager:  GNOME Shell
    Steam Runtime Version:  steam-runtime_0.20200604.0

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 1060 with Max-Q Design/PCIe/SSE2
    Driver Version:  4.6.0 NVIDIA 440.100
    OpenGL Version: 4.6
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x10de
    DeviceID:  0x1c20
    Revision Not Detected
    Number of Monitors:  1
    Number of Logical Video Cards:  2
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 13.54" x 7.64" (15.51" diag)
                                            34.4cm x 19.4cm (39.4cm diag)
    Primary VRAM: 6144 MB

Sound card:
    Audio device: Realtek ALC3246

Memory:
    RAM:  15877 Mb

VR Hardware:
    VR Headset: None detected

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Total Hard Disk Space Available:  117457 Mb
    Largest Free Hard Disk Block:  8149 Mb

Steam Linux Runtime Information:
{
  "can-write-uinput" : true,
  "steam-installation" : {
    "path" : "/home/thiras/.local/share/Steam",
    "data_path" : "/home/thiras/.local/share/Steam",
    "issues" : [
    ]
  },
  "runtime" : {
    "path" : "/home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime",
    "version" : "0.20200604.0",
    "issues" : [
    ],
    "pinned_libs_32" : {
      "list" : [
        "  2000179      4 drwxrwxr-x   2 thiras   thiras       4096 Aug 11 05:18 pinned_libs_32",
        "  1968809      0 -rw-rw-r--   1 thiras   thiras          0 Aug 11 05:18 pinned_libs_32/has_pins",
        "  1968823      4 lrwxrwxrwx   1 thiras   thiras        100 Aug 11 05:18 pinned_libs_32/libGLU.so.1 -> /home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libGLU.so.1.3.08004",
        "  1968826      4 -rw-rw-r--   1 thiras   thiras         80 Aug 11 05:18 pinned_libs_32/system_libvulkan.so.1",
        "  1968808      4 -rw-rw-r--   1 thiras   thiras         80 Aug 11 05:18 pinned_libs_32/system_libdbus-1.so.3",
        "  1968824      4 -rw-rw-r--   1 thiras   thiras         72 Aug 11 05:18 pinned_libs_32/system_libGLU.so.1",
        "  1968804      4 lrwxrwxrwx   1 thiras   thiras         95 Aug 11 05:18 pinned_libs_32/libdbus-1.so.3 -> /home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/lib/i386-linux-gnu/libdbus-1.so.3.5.8",
        "  1968825      4 lrwxrwxrwx   1 thiras   thiras        101 Aug 11 05:18 pinned_libs_32/libvulkan.so.1 -> /home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/libvulkan.so.1.2.135"
      ]
    },
    "pinned_libs_64" : {
      "list" : [
        "  2000180      4 drwxrwxr-x   2 thiras   thiras       4096 Aug 11 05:18 pinned_libs_64",
        "  1968812      0 -rw-rw-r--   1 thiras   thiras          0 Aug 11 05:18 pinned_libs_64/has_pins",
        "  1968810      4 lrwxrwxrwx   1 thiras   thiras        102 Aug 11 05:18 pinned_libs_64/libGLU.so.1 -> /home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libGLU.so.1.3.08004",
        "  1968814      4 -rw-rw-r--   1 thiras   thiras         97 Aug 11 05:18 pinned_libs_64/system_libdbusmenu-glib.so.4",
        "  1968820      4 -rw-rw-r--   1 thiras   thiras         84 Aug 11 05:18 pinned_libs_64/system_libvulkan.so.1",
        "  1968813      4 lrwxrwxrwx   1 thiras   thiras        109 Aug 11 05:18 pinned_libs_64/libdbusmenu-glib.so.4 -> /home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libdbusmenu-glib.so.4.0.13",
        "  1968811      4 -rw-rw-r--   1 thiras   thiras         76 Aug 11 05:18 pinned_libs_64/system_libGLU.so.1",
        "  1968818      4 -rw-rw-r--   1 thiras   thiras         90 Aug 11 05:18 pinned_libs_64/system_libpangox-1.0.so.0",
        "  1968815      4 lrwxrwxrwx   1 thiras   thiras         99 Aug 11 05:18 pinned_libs_64/libcurl.so.4 -> /home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libcurl.so.4.2.0",
        "  1968816      4 -rw-rw-r--   1 thiras   thiras         78 Aug 11 05:18 pinned_libs_64/system_libcurl.so.4",
        "  1968817      4 lrwxrwxrwx   1 thiras   thiras        108 Aug 11 05:18 pinned_libs_64/libpangox-1.0.so.0 -> /home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libpangox-1.0.so.0.3000.0",
        "  1968827      0 lrwxrwxrwx   1 thiras   thiras         12 Aug 11 05:18 pinned_libs_64/libcurl.so.3 -> libcurl.so.4",
        "  1968819      4 lrwxrwxrwx   1 thiras   thiras        103 Aug 11 05:18 pinned_libs_64/libvulkan.so.1 -> /home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libvulkan.so.1.2.135",
        "  1968822      4 -rw-rw-r--   1 thiras   thiras         95 Aug 11 05:18 pinned_libs_64/system_libdbusmenu-gtk.so.4",
        "  1968821      4 lrwxrwxrwx   1 thiras   thiras        108 Aug 11 05:18 pinned_libs_64/libdbusmenu-gtk.so.4 -> /home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/libdbusmenu-gtk.so.4.0.13"
      ]
    }
  },
  "os-release" : {
    "id" : "ubuntu",
    "id_like" : [
      "debian"
    ],
    "name" : "Ubuntu",
    "pretty_name" : "Ubuntu 20.04.1 LTS",
    "version_id" : "20.04",
    "version_codename" : "focal"
  },
  "container" : {
    "type" : "none"
  },
  "driver_environment" : [
    "__GLX_VENDOR_LIBRARY_NAME=nvidia"
  ],
  "architectures" : {
    "i386-linux-gnu" : {
      "can-run" : true,
      "library-issues-summary" : [
      ],
      "graphics-details" : {
        "x11/vulkan" : {
          "messages" : "ERROR: [Loader Message] Code 0 : /usr/lib/x86_64-linux-gnu/libvulkan_radeon.so: wrong ELF class: ELFCLASS64\nERROR: [Loader Message] Code 0 : /usr/lib/x86_64-linux-gnu/libvulkan_intel.so: wrong ELF class: ELFCLASS64\nERROR: [Loader Message] Code 0 : /home/thiras/.local/share/Steam/ubuntu12_64/steamoverlayvulkanlayer.so: wrong ELF class: ELFCLASS64\n",
          "renderer" : "GeForce GTX 1060 with Max-Q Design",
          "version" : "1.1.119 (device 10de:1c20) (driver 440.400.0)"
        },
        "glx/gl" : {
          "renderer" : "GeForce GTX 1060 with Max-Q Design/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.100",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "messages" : "Waffle error: 0x1 WAFFLE_ERROR_FATAL: dlopen(\"libEGL.so.1\") failed: libEGL.so.1: wrong ELF class: ELFCLASS64\n",
          "renderer" : null,
          "version" : null,
          "library-vendor" : "unknown",
          "issues" : [
            "cannot-load"
          ],
          "exit-status" : 1
        },
        "egl_x11/glesv2" : {
          "messages" : "Waffle error: 0x1 WAFFLE_ERROR_FATAL: dlopen(\"libEGL.so.1\") failed: libEGL.so.1: wrong ELF class: ELFCLASS64\n",
          "renderer" : null,
          "version" : null,
          "library-vendor" : "unknown",
          "issues" : [
            "cannot-load"
          ],
          "exit-status" : 1
        }
      },
      "dri_drivers" : [
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/i915_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/i965_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/iris_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/kms_swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/nouveau_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/nouveau_vieux_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/r200_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/r300_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/r600_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/radeon_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/radeonsi_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/virtio_gpu_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/vmwgfx_dri.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/zink_dri.so"
        }
      ],
      "va-api_drivers" : [
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/i965_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/iHD_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/nouveau_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/r600_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/dri/radeonsi_drv_video.so"
        },
        {
          "library_path" : "/home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/i386-linux-gnu/dri/dummy_drv_video.so"
        }
      ],
      "vdpau_drivers" : [
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_nouveau.so",
          "library_link" : "libvdpau_nouveau.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_nouveau.so.1",
          "library_link" : "libvdpau_nouveau.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_nvidia.so",
          "library_link" : "libvdpau_nvidia.so.440.100"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_nvidia.so.1",
          "library_link" : "libvdpau_nvidia.so.440.100"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_r300.so",
          "library_link" : "libvdpau_r300.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_r300.so.1",
          "library_link" : "libvdpau_r300.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_r600.so",
          "library_link" : "libvdpau_r600.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_r600.so.1",
          "library_link" : "libvdpau_r600.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_radeonsi.so",
          "library_link" : "libvdpau_radeonsi.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_radeonsi.so.1",
          "library_link" : "libvdpau_radeonsi.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/i386-linux-gnu/vdpau/libvdpau_trace.so.1",
          "library_link" : "libvdpau_trace.so.1.0.0"
        }
      ],
      "glx_drivers" : [
        {
          "library_soname" : "libGLX_indirect.so.0",
          "library_path" : "/usr/lib/i386-linux-gnu/libGLX_mesa.so.0.0.0"
        },
        {
          "library_soname" : "libGLX_mesa.so.0",
          "library_path" : "/usr/lib/i386-linux-gnu/libGLX_mesa.so.0.0.0"
        },
        {
          "library_soname" : "libGLX_nvidia.so.0",
          "library_path" : "/usr/lib/i386-linux-gnu/libGLX_nvidia.so.440.100"
        }
      ]
    },
    "x86_64-linux-gnu" : {
      "can-run" : true,
      "library-issues-summary" : [
      ],
      "graphics-details" : {
        "x11/vulkan" : {
          "messages" : "ERROR: [Loader Message] Code 0 : /usr/lib/i386-linux-gnu/libvulkan_intel.so: wrong ELF class: ELFCLASS32\nERROR: [Loader Message] Code 0 : /usr/lib/i386-linux-gnu/libvulkan_radeon.so: wrong ELF class: ELFCLASS32\nERROR: [Loader Message] Code 0 : /home/thiras/.local/share/Steam/ubuntu12_32/steamoverlayvulkanlayer.so: wrong ELF class: ELFCLASS32\n",
          "renderer" : "GeForce GTX 1060 with Max-Q Design",
          "version" : "1.1.119 (device 10de:1c20) (driver 440.400.0)"
        },
        "glx/gl" : {
          "renderer" : "GeForce GTX 1060 with Max-Q Design/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.100",
          "library-vendor" : "glvnd"
        },
        "egl_x11/gl" : {
          "renderer" : "GeForce GTX 1060 with Max-Q Design/PCIe/SSE2",
          "version" : "4.6.0 NVIDIA 440.100",
          "library-vendor" : "glvnd"
        },
        "egl_x11/glesv2" : {
          "renderer" : "GeForce GTX 1060 with Max-Q Design/PCIe/SSE2",
          "version" : "OpenGL ES 3.2 NVIDIA 440.100",
          "library-vendor" : "glvnd"
        }
      },
      "dri_drivers" : [
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/i915_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/i965_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/iris_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/kms_swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/nouveau_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/nouveau_vieux_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/r200_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/r300_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/r600_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/radeon_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/radeonsi_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/swrast_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/virtio_gpu_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/vmwgfx_dri.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/zink_dri.so"
        }
      ],
      "va-api_drivers" : [
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/i965_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/nouveau_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/r600_drv_video.so"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/dri/radeonsi_drv_video.so"
        },
        {
          "library_path" : "/home/thiras/.local/share/Steam/ubuntu12_32/steam-runtime/usr/lib/x86_64-linux-gnu/dri/dummy_drv_video.so"
        }
      ],
      "vdpau_drivers" : [
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_nouveau.so",
          "library_link" : "libvdpau_nouveau.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_nouveau.so.1",
          "library_link" : "libvdpau_nouveau.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_nvidia.so",
          "library_link" : "libvdpau_nvidia.so.440.100"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_nvidia.so.1",
          "library_link" : "libvdpau_nvidia.so.440.100"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_r300.so",
          "library_link" : "libvdpau_r300.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_r300.so.1",
          "library_link" : "libvdpau_r300.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_r600.so",
          "library_link" : "libvdpau_r600.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_r600.so.1",
          "library_link" : "libvdpau_r600.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_radeonsi.so",
          "library_link" : "libvdpau_radeonsi.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_radeonsi.so.1",
          "library_link" : "libvdpau_radeonsi.so.1.0.0"
        },
        {
          "library_path" : "/usr/lib/x86_64-linux-gnu/vdpau/libvdpau_trace.so.1",
          "library_link" : "libvdpau_trace.so.1.0.0"
        }
      ],
      "glx_drivers" : [
        {
          "library_soname" : "libGLX_indirect.so.0",
          "library_path" : "/usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0.0.0"
        },
        {
          "library_soname" : "libGLX_mesa.so.0",
          "library_path" : "/usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0.0.0"
        },
        {
          "library_soname" : "libGLX_nvidia.so.0",
          "library_path" : "/usr/lib/x86_64-linux-gnu/libGLX_nvidia.so.440.100"
        }
      ]
    }
  },
  "locale-issues" : [
  ],
  "locales" : {
    "<default>" : {
      "resulting-name" : "en_US.UTF-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    },
    "C" : {
      "resulting-name" : "C",
      "charset" : "ANSI_X3.4-1968",
      "is_utf8" : false
    },
    "C.UTF-8" : {
      "resulting-name" : "C.UTF-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    },
    "en_US.UTF-8" : {
      "resulting-name" : "en_US.UTF-8",
      "charset" : "UTF-8",
      "is_utf8" : true
    }
  },
  "egl" : {
    "icds" : [
      {
        "json_path" : "/usr/share/glvnd/egl_vendor.d/10_nvidia.json",
        "library_path" : "libEGL_nvidia.so.0"
      },
      {
        "json_path" : "/usr/share/glvnd/egl_vendor.d/50_mesa.json",
        "library_path" : "libEGL_mesa.so.0"
      }
    ]
  },
  "vulkan" : {
    "icds" : [
      {
        "json_path" : "/usr/share/vulkan/icd.d/radeon_icd.x86_64.json",
        "library_path" : "/usr/lib/x86_64-linux-gnu/libvulkan_radeon.so",
        "api_version" : "1.2.128"
      },
      {
        "json_path" : "/usr/share/vulkan/icd.d/intel_icd.i686.json",
        "library_path" : "/usr/lib/i386-linux-gnu/libvulkan_intel.so",
        "api_version" : "1.2.131"
      },
      {
        "json_path" : "/usr/share/vulkan/icd.d/nvidia_icd.json",
        "library_path" : "libGLX_nvidia.so.0",
        "api_version" : "1.1.119"
      },
      {
        "json_path" : "/usr/share/vulkan/icd.d/radeon_icd.i686.json",
        "library_path" : "/usr/lib/i386-linux-gnu/libvulkan_radeon.so",
        "api_version" : "1.2.128"
      },
      {
        "json_path" : "/usr/share/vulkan/icd.d/intel_icd.x86_64.json",
        "library_path" : "/usr/lib/x86_64-linux-gnu/libvulkan_intel.so",
        "api_version" : "1.2.131"
      }
    ]
  },
  "desktop-entries" : [
    {
      "id" : "steam.desktop",
      "commandline" : "/usr/bin/steam %U",
      "filename" : "/usr/share/applications/steam.desktop",
      "default_steam_uri_handler" : true,
      "steam_uri_handler" : true
    }
  ],
  "cpu-features" : {
    "x86-64" : true,
    "sse3" : true,
    "cmpxchg16b" : true
  }
}

The Steam Linux Runtime container tool is not installed.
```

</details>

I was using the latest WINE from WINE HQ official repository (5.0.1) and latest winetricks.

The game launcher stars without any glitch or something after @ldenti 's guide. The game crashes just after loading screen appears and plays few seconds of music. The loading mouse icon also appears and moves as expected.

I've also tried to follow @tobiasheineken 's points but my Library list at the winecfg doesn't have devenum.dll nor quartz.dll. I've tried to install directmusic and/or quartz and tried both those DLL enabled and both disabled/removed.

**Comment by mozo78** [comment]
Latest Wine is 5.15.

**Comment by thiras** [comment]
@mozo78 latest `stable` is 5.0.2 now. 5.15 is development release.

**Comment by mozo78** [comment]
Try with 5.15, there are many improvements since 5.0.

**Comment by thiras** [comment]
> Try with 5.15, there are many improvements since 5.0.

@mozo78 Mate we are trying to run it through Proton not through pure WINE. You seem like have no idea what we are trying here.

**Comment by mozo78** [comment]
> I was using the latest WINE from WINE HQ official repository (5.0.1) and latest winetricks.

> @mozo78 Mate we are trying to run it through Proton not through pure WINE. You seem like have no idea what we are trying here.

Yeah :)
And you can use Steam games with Vanilla wine if you don't know by accident.
And I know very well what you are doing, believe me:
https://www.youtube.com/c/NikolayStefanovkaminatadotnet

**Comment by thiras** [comment]
@mozo78 I know. This is Proton repository. I'd report the issue WINE repo otherwise.

**Comment by mozo78** [comment]
I just suggest to try it to know if the problem exists on Wine too.

**Comment by thiras** [comment]
> I just suggest to try it to know if the problem exists on Wine too.

Already working on it. Will report if situation changes.

**Comment by mineturtle36** [comment]
> > I just suggest to try it to know if the problem exists on Wine too.
> 
> Already working on it. Will report if situation changes.

insert 'well, we're waiting' meme here

Any luck?

**Comment by thiras** [comment]
@mineturtle36 unfortunately no luck. Still looking for a solution. This is the greatest game ever made for WW2, very unfortunate for us.

**Comment by thiras** [comment]
The error is related to d3dcompiler.

`fixme:d3dcompiler:compile_shader Compilation target "fx_2_0" not yet supported` found in the logs just before the crash.

Winetricks d3dcompiler_42/43/47 didn't help with the proton.

But I've managed to run the GOG version in 32-bit prefix with [this report](https://appdb.winehq.org/objectManager.php?sClass=version&iId=28092&iTestingId=89051) I just add d3dcompiler_43 winetrick after the guide.

I guess 32-bit prefix might be a must to run the game properly. #449 would help really.

**Comment by kisak-valve** [comment]
Thanks @thiras. `fixme:d3dcompiler:compile_shader Compilation target "fx_2_0" not yet supported` points towards https://bugs.winehq.org/show_bug.cgi?id=37676.

**Comment by mineturtle36** [comment]
I tried again after the Steam Deck has been around for a while now - and Hearts of Iron III (Vanilla with all expansions) starts and works without problems. I tried to crash it by torturing the game engine a bit - but couldn't do it. Awesome!

protontricks d3dx9 and d3dcompiler_42 are mandatory, otherwise we crash on startup (same as before). This is mentioned by multiple users over on https://www.protondb.com/app/25890

This bug can be closed I guess.

However, there is a performance problem. When loading the list of commanders the game seems like it is loading the whole list every time from scratch - each time taking ~5 seconds - which gets very annoying very fast.
On Windows on the other hand it takes ~5 seconds only the first time, and then it is seemingly cached somewhere - resulting in almost instant reaction for the rest of that session.

I have tried it on a 32bit prefix too (with dotnet20 installed) but the same issue exists there.
The workaround is to delete ALL leader images from */Hearts of Iron 3/gfx/pictures/portraits/ - which is 17697 assets wasted.

There must be a better way. Should I open a new bug?

**Comment by kisak-valve** [comment]
Hello @mineturtle36, we're using one issue report per unofficially supported game title. This is the right place for all feedback related to running Hearts of Iron III with Proton.
